### PR TITLE
feat: Update user-info

### LINKED ISSUE #44: Update user info card
### **Objective:**

There was already a component called "userInfo" however, it should be updated/changed as stated below. 

### **Description:**
* In the already existing "components/UserInfo.tsx" component, update the component to display the user's names and username
* Update the user's type definition in the `types/index.ts`to include `username` property.
* Update each user in `data/users.ts` to have a username to facilitate testing.
* The user's names remain unchanged.
* The user's username should be preceded with an "@", and should have a different font family as the name.
* Update the `Friend`component to contain the `<UserInfo/>` card instead of a plain name.

### **Acceptance Criteria:**.

* The component matches the design.
* The component is used in the friends list. 

### **Design:**

[Figma](https://www.figma.com/design/ZcPlin9PqvfVKaeM8VfUov/SocialApp?node-id=1454-2578&node-type=frame&t=BNModlzBCgtwumfQ-11)
Image

### PR DESCRIPTION
**Pull request by Odile078**
closes #44

### CODE REVIEW

File: components/UserInfo.tsx [hunk 1]
```diff
@@ -41,13 +36,14 @@ const styles = StyleSheet.create({
...
   },
-  detailsContainer: {
-    flexDirection: "row",
-    alignItems: "center",
-    gap: 6,
-    marginTop: 2,
+  username: {
+    fontSize: 15,
+    lineHeight: 20,
+    color: "#000000",
```
Comment (serapieTuyishime) [inline components/UserInfo.tsx:46]: Use theme

File: data/friends.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ export const visibleFriends = [
     id: "u1",
     firstName: "Erin",
     lastName: "Wagner",
+    username: "@erinwagner",
```
Comment (serapieTuyishime) [inline data/friends.ts:6]:
```suggestion
    username: "erinwagner",
```

File: data/friends.ts
Comment (serapieTuyishime) [inline data/friends.ts:1]: As its unlikely that the database with store the username with a preceding "@", its better that we hardcode it in the `<UserInfo/>` component

File: components/UserInfo.tsx [hunk 3]
```diff
@@ -1,37 +1,32 @@
...
-        fontFamily="marfa"
-        fontWeight="medium"
+        style={styles.name}
+        fontWeight="semibold"
       >
-        {`${firstName} ${lastName}`}
+       {firstName} {lastName}
+      </CustomText>
+      <CustomText size="sm" fontFamily="writer-mono" style={styles.username}>
+        {username}
```
Comment (serapieTuyishime) [inline components/UserInfo.tsx:28]:
```suggestion
        @{username}
```

File: data/users.ts
Comment (serapieTuyishime) [inline data/users.ts:1]: Same for this file, remove the preceding "@" on the usernames

**Review by serapieTuyishime — CHANGES_REQUESTED** [review]

**Review by serapieTuyishime — APPROVED** [review]

**Review by mhenriette — APPROVED** [review]

**Review by AnasAlamin — APPROVED** [review]

**Pull request merged**